### DISCUSSION TITLE
How is lsp.json generated?

### DISCUSSION THREAD

**Discussion by predragnikolic** [microsoft/lsprotocol#49] [Q&A] [answered]
Hello :wave: ,
cool project :)

#1 How was the the https://github.com/microsoft/lsprotocol/blob/main/generator/lsp.json generated?

Is there a place where it was just downloaded
or was it written by hand 
or something third? :)

**Comment by karthiknadig** [comment, marked as answer]
@predragnikolic The lsp.json is generated from the typescript implementation of LSP for node. We pull it down from the LSP node repository. See here:
https://github.com/microsoft/lsprotocol/blob/598f2314e0ee4d48e421c1e795ca822fc95d368e/noxfile.py#L54-L56

**Reply by predragnikolic** [reply]
Thanks 🙂
